Night mode version for docs

pdilyard opened the pull request:
This could probably use some cleaning up, but this is a good starting point. Here are some screenshots:

![](http://puu.sh/lEo15/3d60cd3feb.png)

![](http://puu.sh/lEo6n/34107cf960.png)

![](http://puu.sh/lEoc8/f808fce6b4.png)

Feel free to pull it down and play with it, I'd love to hear any feedback.

Solves #474

josevalim commented:
Thank you @pdilyard, this looks great! <3 Some points to discuss:

1. Can we make the font less white and a bit more greyish? I am afraid the white is a bit too strong right now and won't be beneficial on night mode. Tim's script had this: https://pbs.twimg.com/media/CTqMbN5UAAAqO0b.png:large

2. I am not convinced about the "day | night" switch at the top. It is a command that won't be triggered frequently. I was wondering we could have it at the footer of the page. Something like:

```
Built using ExDoc (v0.10.0),  designed by Friedel Ziegelmayer.
              Choose: Day mode | Night mode
```

However, please don't act on this feedback yet. Let me know your opinion and let's also wait for @Dignifiedquire's input.

Everything else looks excellent to me. Great use of local storage!

pdilyard commented:
Thanks for the quick response @josevalim.

1. You're probably right about the colors not being beneficial in night mode. I went with as many predefined colors as possible, but I could definitely add a special color palette for the night fonts.

2. I really like the idea of the switch in the footer a lot better than the button I have now.

milmazz commented on assets/less/night/content/functions.less:
```diff
@@ -0,0 +1,57 @@
+
+  }
+
+  h2 {
```
This block can be deleted.

pdilyard replied:
Oops, forgot to take all these extras out :smile:

milmazz commented on assets/less/night/content/functions.less:
```diff
@@ -0,0 +1,57 @@
+
+  }
+
+  h3 {
```
This block can be deleted.

milmazz commented on assets/less/night/content/functions.less:
```diff
@@ -0,0 +1,57 @@
+
+  }
+
+  h4 {
```
This block can be deleted.

milmazz commented on assets/less/night/content/functions.less:
```diff
@@ -0,0 +1,57 @@
+
+  }
+
+  h5 {
```
This block can be deleted.

milmazz commented on assets/less/night/content/general.less:
```diff
@@ -0,0 +1,10 @@
+h1, h2, h3, h4, h5, h6 {
```
Just to be consistent with the rest of the code, please, use one line per header:

```css
h1,
h2,
h3,
h4,
h5,
h6 {
...
}
```

milmazz commented on assets/less/night/content/functions.less:
```diff
@@ -0,0 +1,57 @@
+  .specs-list {
+
+    pre {
+
```
:scissors: Extra space

milmazz commented on assets/less/variables.less:
```diff
@@ -17,6 +17,12 @@
+@nightCodeBlue: #4B556E;
+@nightMediumBlue: #69778C;
+@nightPurple: #CEB4ED;
+
```
:scissors:

pdilyard replied:
Is this comment a mistake? I'm using those variables. Or are you suggesting I move them to a variables.less file inside the night folder?

josevalim replied:
I think it was regarding the extra new line or whitespace :D

milmazz replied:
Sorry, I just meant to "delete the extra space" :)

pdilyard commented:
@Dignifiedquire ping

Dignifiedquire commented:
will take a look in the morning

Dignifiedquire commented:
@pdilyard thanks for this, some feedback below:

* Switch, it looks great, but the position is not great, so if we can just move it as it is into the footer?
* Colors, we definitely need to change some things here. Most important are
  * Code border and background
  * Code syntax theme should be one that's designed for dark mode (I suggest tomorrow night from highlight.js)
  * Text color in general (probably should be lighter)
  * Function header background

pdilyard commented:
Alright here are some updates of where I'm at with @Dignifiedquire's feedback so far.

- I changed the syntax theme to `tomorrow-night` in night mode.
- I changed the background of the code blocks to a dark gray that works a little better with the syntax highlighting.
- I changed the overall text color to a less-harsh gray.
- I ended up just moving the switch to the footer as a text link. I tried to move the icon-based switch as it was to the footer, but I couldn't figure out a good way to position it that I felt was easy to find.

Note the new text colors:
![](http://puu.sh/lJkxT/f2001bc993.png)

Note the new toggle switch in the footer, as well as the new code blocks:
![](http://puu.sh/lJkyI/85b776f4ee.png)

I also have a question for @Dignifiedquire:
- What were you thinking for the function header backgrounds? Should those be more gray? I want to make sure they stand out, as discussed in #487.

Any other feedback is much appreciated :smile:

josevalim commented:
This looks great. Once again, let's wait for @Dignifiedquire to sign off and merge this. :) We should do a release right after! This will be great with the other style changes.

Dignifiedquire commented:
Sorry guys for being so absent, checking the branch out now

Dignifiedquire commented:
@pdilyard are you sure you pushed your changes, pulling down your branch there seem to be no changes

pdilyard commented:
@Dignifiedquire I don't know what just happened here. For some reason git thought I needed to pull down my feature branch and gave me a bunch of merge conflicts on it, even though the feature branch has only ever had changes made by me directly. I resolved the conflicts, and it threw in a bunch of repeat commits. Let me know if you have a tip on how I can clean up this history a bit.

josevalim commented:
@Dignifiedquire nothing to be sorry about :D I mostly mention you to be make it clear to @pdilyard why we are waiting. It is OK to be absent or to tell us to wait for a week or even to tell us to do as we please if you can't take a look at it soon. :)

josevalim commented:
@pdilyard it is ok if the branch is a bit messy, I can squash everything before merging. Just make sure there is no spurious code in the diff here on Github and we are fine. :)

pdilyard commented:
@josevalim thanks, the diff looks good :+1:

pdilyard commented:
But anyway, @Dignifiedquire the changes should be there for you now

Dignifiedquire commented:
Alright, I think this is in a really good state now, only thing I'm not a 100% happy with is the function header background and the text background, but I don't have any good ideas right now I'm afraid

josevalim commented:
Ok. I am going to merge and do a new release today. :) We will wait for more feedback and continue improving when it comes to the background colors.

josevalim commented:
Thank you @pdilyard and @Dignifiedquire!  :heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
